torch/csrc/jit/codegen/cuda/glfdc/eval_vm.h

```diff
@@ -0,0 +1,734 @@
+#pragma once
+
+#include <torch/csrc/jit/codegen/cuda/glfdc/eval_uinstr.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+namespace glfdc {
+
+using MemoryBlock = DEVector<scalar_type>;
+
+template <typename BindingFn_>
+c10::optional<scalar_type> evaluateLoop(
+    const std::vector<UInst>& instructions,
+    const InitialStack<scalar_type>& init,
+    MemoryBlock& memory,
+    BindingFn_& fn,
+    EvalState& es) noexcept {
+  // Instruction pointer
+  unsigned ip = 0;
+
+  // Registers
+  scalar_type a, b, t;
+
+  // Stack of operands
+  OperandsStack operands(memory);
+  // Copy values to operand stack
+  init.copy_to(operands);
+
+  ScratchSpace tmps(memory);
+  bool recalled_flag = false;
+
+  // Zero the registers
+  a = 0;
+  b = 0;
+  t = 0;
+
+  for (; ip < instructions.size(); ++ip) {
```

Comment:
It seems like having these 3 registers, a scratch space, and 2 orders of stack visit is creating quite a few copies of the same instructions. Just wondering if we could simplify this instruction set a bit.